src/libraries/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.X509.cs

```diff
@@ -57,7 +57,14 @@ internal static byte[] X509GetRawData(SafeSecCertificateHandle cert)
 
             if (ret == 1)
             {
-                return CoreFoundation.CFGetData(data);
+                try
+                {
+                    return CoreFoundation.CFGetData(data);
+                }
+                finally
+                {
+                    data.Dispose();
```

Comment:
Minor, but I think it would make sense to put this in a `using` so that the safe handle gets disposed regardless if the result succeeds, or not.

```diff
         internal static byte[] X509GetRawData(SafeSecCertificateHandle cert)
         {
             int osStatus;
             SafeCFDataHandle data;
 
             int ret = AppleCryptoNative_X509GetRawData(
                 cert,
                 out data,
                 out osStatus);
 
+            using (data)
+            {
                 if (ret == 1)
                 {
                     return CoreFoundation.CFGetData(data);
                 }
 
                 if (ret == 0)
                 {
                     throw CreateExceptionForOSStatus(osStatus);
                 }
 
                 Debug.Fail($"Unexpected return value {ret}");
                 throw new CryptographicException();
+            }
         }
```

Even if the call fails, we still want to dispose of the safe handle (whose interior pointer is null) so that we don't leave it hanging around in the finalization queue unnecessarily.

Reply:
Sure. I assumed it will be null if ret is not equal to 1. But nothing bad will happen if we do a using on a null value

Reply:
> I assumed it will be null if ret is not equal to 1.

The marshaller won't ever give back a null `SafeCFDataHandle`, even if the native API returns null. Instead, the marshaller will create a new `SafeCFDataHandle` and the `handle` value will be `IntPtr.Zero`.